src/start.clj

```diff
@@ -1,19 +1,33 @@
 (ns start
   (:gen-class)
-  (:require [clj-time.core   :as t])
+  (:require [util]
+            [taoensso.timbre  :as timbre])
   (:import  [org.apache.activemq ActiveMQSslConnectionFactory]
             [javax.jms TextMessage DeliveryMode Session]))
 
 (defn with-push-to-cloud-jms-connection
-  "CALL (use connection destination) for the push-to-cloud JMS queue with URL, QUEUE, USERNAME and PASSWORD."
-  [url queue username password call]
-  (let [factory (new ActiveMQSslConnectionFactory url)]
+  "CALL (use connection destination) for the push-to-cloud JMS queue with ENVIRONMENT."
+  [environment call]
+  (let [vault-path (format "secret/dsde/gotc/%s/activemq/logins/zamboni" environment)
+        {:keys [url username password queue]} (util/vault-secrets vault-path)
+        factory (new ActiveMQSslConnectionFactory url)]
     (with-open [connection (.createQueueConnection factory username password)]
       (call connection queue))))
 
-(defn send-text
-  "Enqueue the TEXT with PROPERTIES map to push-to-cloud JMS queue with URL, QUEUE, USERNAME and PASSWORD."
-  [url queue username password text properties]
```

Comment:
Consuming from a DESTINATION seems odd.
Maybe rename it to SOURCE or just QUEUE?
Maybe create a "durable consumer" and block in RECEIVE?
It might have to run in a different thread later, preserving the main thread to respond to other events.